src/vm_stack.zig

```diff
@@ -3431,16 +3437,16 @@ const InstructionFuncs = struct {
         const memory_offset_u32 = @as(u32, @intCast(memory_offset));
         const length_u32 = @as(u32, @intCast(length));
 
-        var source = data.bytes.items[data_offset_u32 .. data_offset_u32 + length_u32];
-        var destination = buffer[memory_offset_u32 .. memory_offset_u32 + length_u32];
-        std.mem.copy(u8, destination, source);
+        const source = data.bytes.items[data_offset_u32 .. data_offset_u32 + length_u32];
+        const destination = buffer[memory_offset_u32 .. memory_offset_u32 + length_u32];
+        std.mem.copyForwards(u8, destination, source);
```

Comment:
`@memcpy`?

Reply:
@memcpy works for only non-aliasing values. In some of these, they variables alias, so we have to use std.mem.copyForwards. I'll double check which ones alias and which ones don't and use `@memcpy` where possible.